src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -268,25 +268,30 @@ impl<'a, T: State + StateReader> BusinessLogicSyscallHandler<'a, T> {
         Ok(SyscallResponse { gas, body })
     }
 
+    fn constructor_entry_points(&self, contract_class: CompiledClass) -> Result<bool, StateError> {
+        match contract_class {
+            CompiledClass::Deprecated(class) => Ok(class
+                .entry_points_by_type
+                .get(&EntryPointType::Constructor)
+                .ok_or(ContractClassError::NoneEntryPointType)?
+                .is_empty()),
+            CompiledClass::Casm(class) => Ok(class.entry_points_by_type.constructor.is_empty()),
+        }
+    }
+
     fn execute_constructor_entry_point(
         &mut self,
         contract_address: &Address,
         class_hash_bytes: ClassHash,
         constructor_calldata: Vec<Felt252>,
         remainig_gas: u128,
     ) -> Result<CallResult, StateError> {
-        let contract_class: ContractClass = self
+        let contract_class: CompiledClass = self
```

Comment:
Is this type annotation needed?

Reply:
it was required do to the `try_into` call, but we deleted it.